File: qqbot-persistence/src/main/java/com/mieai/qqbot/persistence/plugin/PluginDeliveryStatus.java
package com.mieai.qqbot.persistence.plugin;

public enum PluginDeliveryStatus {
    PENDING, IN_PROGRESS, RETRY_WAIT, SUCCEEDED, DEAD_LETTER, PAUSED;

    public boolean terminal() {
        return this == SUCCEEDED || this == DEAD_LETTER;
    }
}

File: qqbot-persistence/src/main/kotlin/com/mieai/qqbot/persistence/plugin/PluginDeliveryStatus.kt
package com.mieai.qqbot.persistence.plugin

enum class PluginDeliveryStatus {
    PENDING,
    IN_PROGRESS,
    RETRY_WAIT,
    SUCCEEDED,
    DEAD_LETTER,
    PAUSED,
    ;

    fun terminal(): Boolean = this == SUCCEEDED || this == DEAD_LETTER
}
